v2/internal/io/size.go

```diff
@@ -0,0 +1,19 @@
+package io
+
+import "io"
+
+// Size returns the total number of readable bytes in r, discarding any bytes read.
+func Size(r io.Reader) (uint64, error) {
```

Comment:
I don't think you need this; you could just use `io.Copy` with `io.Discard`, which gives you a total `n` of bytes copied.

Reply:
aha thank you 🍻